src/user/components/my_details/buttons.js

```diff
@@ -1,24 +1,11 @@
 const React = require('react')
 
-const Buttons = (
-  { edit_member_click
-  , cancel_member_click
-  , mode
-  }
-) =>
+const Buttons = ({ mode, ...props }) =>
   mode === 'view'
-    ? <ViewButton edit_member_click={edit_member_click} />
-    : <EditButtons
-        {...
-          { cancel_member_click
-          }
-        }
-      />
+    ? <ViewButton {...props}/>
```

Comment:
While this is neater, I think it's better to not to pass the props that aren't needed through. Edit buttons could be neatened up though: `<EditButtons cancel_member_click=cancel_member_click>`